### ISSUE TITLE
Enable drag/drop on table columns

### ISSUE THREAD

**Issue by tlambert03** [napari/magicgui#102]
After #61, it would be nice to enable drag-and-drop to reorder table columns.  This is possible (in qt) with `self._qwidget.horizontalHeader().setSectionsMovable(True)` but it's a bit tricky because the internal table model in the widget doesn't actually reorder the column headers, it simply updates the `horizontalHeader().visualIndex()` for each column index.  Doable, but better off in a separate PR
